Base.rte/Devices/Tools/Constructor/Constructor.lua

```diff
@@ -266,69 +295,99 @@ function Update(self)
 				self.toautobuild = false;
 			end
 
-			if self.Sharpness == 0 then
-
-				-- digging
-				if ToActor(actor):GetController():IsState(Controller.WEAPON_FIRE) then
-
-					local angle = ToActor(actor):GetAimAngle(true);
+			if self.Sharpness == 0 then	-- dig/spray
```

Comment:
I assume most of what you did here was little fixes. I don't really wanna read through the whole thing but I will say
1. It should be split into functions
2. It shouldn't use sharpness. That's a 2010 hack and we ain't in 2010 anymore.

Reply:
A case of working with what was there already. Eliminating sharpness hacks across all data will probably be a PR of its' own. :P

Reply:
Yeah, totally fair. It'll have to be done sometime but it is a lot of work, and keeping things in separate prs is helpful to (theoretically) speed up review time.